src/character.cpp

```diff
@@ -191,17 +193,19 @@ bool Character::move_effects(bool attacking)
             remove_effect("in_pit");
         }
     }
-    if (has_effect("grabbed")){
+    if (has_effect("grabbed") && !attacking){
         int zed_number = 0;
         for( auto &&dest : g->m.points_in_radius( pos(), 1, 0 ) ){
-            if (g->mon_at(dest) != -1){
+            if (g->mon_at(dest) != -1 && g->zombie(g->mon_at(dest)).has_flag(MF_GRABS)) {
```

Comment:
Some monsters don't have MF_GRAB, they have a grab special attack, so this results in grabs from those monsters not working now.
Test case: make yourself invincible (just to make it less annoying), spawn a bunch of regular zombies.  After one or more establishes a grab, step away, it will automatically work.